src/nmod_poly/test/t-evaluate_geometric_nmod_vec_fast.c

```diff
@@ -0,0 +1,67 @@
+/*
+    Copyright (C) 2025, Vincent Neiger, Éric Schost
+    Copyright (C) 2025, Mael Hostettler
+
+    This file is part of FLINT.
+
+    FLINT is free software: you can redistribute it and/or modify it under
+    the terms of the GNU Lesser General Public License (LGPL) as published
+    by the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.  See <https://www.gnu.org/licenses/>.
+*/
+
+#include "test_helpers.h"
+#include "nmod_vec.h"
+#include "nmod_poly.h"
+#include "nmod.h"
+
+TEST_FUNCTION_START(nmod_poly_evaluate_geometric_nmod_vec_fast, state)
+{
+    int i, result = 1;
+
+    for (i = 0; i < 1000 * flint_test_multiplier(); i++)
+    {
+        nmod_poly_t P;
+        nn_ptr y, z;
+        ulong mod, r;
+        slong n, npoints;
+
+        npoints = (i < 10) ? i : n_randint(state, 100);
```

Comment:
```suggestion
        npoints = (i < 10) ? i : n_randint(state, 500);
```
(so that it is less rare that in tests the degree `n` is less than npoints)